mlir/include/Driver/Pipelines.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "mlir/Pass/Pass.h"
+
+namespace catalyst {
+namespace driver {
+
+void createEnforceRuntimeInvariantsPipeline(mlir::OpPassManager &pm);
+void createHloLoweringPipeline(mlir::OpPassManager &pm);
+void createQuantumCompilationPipeline(mlir::OpPassManager &pm);
+void createBufferizationPipeline(mlir::OpPassManager &pm);
+void createLLVMDialectLoweringPipeline(mlir::OpPassManager &pm);
+void createDefaultCatalystPipeline(mlir::OpPassManager &pm);
+
+void registerEnforceRuntimeInvariantsPipeline();
+void registerHloLoweringPipeline();
+void registerQuantumCompilationPipeline();
+void registerBufferizationPipeline();
+void registerLLVMDialectLoweringPipeline();
+void registerDefaultCatalystPipeline();
+void registerAllCatalystPipelines();
+
+/// Pipeline descriptor
+struct Pipeline {
+    using Name = std::string;
```

Comment:
If it's just used in one place here and the typedef is this simple, it might be better to just do
```C
std::string name;
llvm::SmallVector<std::string> passes;
```

to increase readability (as a developer won't need to keep track of a `using`), since the new names given to the `using` types really don't add new information on top of what the variable names can already convey.

Reply:
Speaking of which, can we name it `pipelineName`? So not to be confused with pass names.

Reply:
For the name, I think it will be clear since it will now be pipeline.getName() instead of calling pipeline.name.

Reply:
Good point!